waterlog/src/Components/Map.js

```diff
@@ -27,7 +30,95 @@ const backgroundColor = '#253238';
 const errorColor = '#FF1744';
 const lighterColor = '#4F5B62';
 
-export default class MapComponent extends Component {
+function formatMapData(data) {
+  let markers, segments;
+  let todaysLeaks = data.leaks.filter(leak => (
+    (new Date(leak.latestTimeStamp)).getDay() === (new Date(Date.now())).getDay() &&
+    (new Date(leak.latestTimeStamp)).getMonth() === (new Date(Date.now())).getMonth() &&
+    (new Date(leak.latestTimeStamp)).getFullYear() === (new Date(Date.now())).getFullYear()
+  ));
+
+  segments = data.segments.map(seg => {
+    let leak = todaysLeaks.find(leak => leak.segmentsId === seg.id);
+    if (leak) {
+      seg.status = (leak.resolvedStatus.toLowerCase() === "unresolved") ? "leak" : "normal";
+    } else {
+      seg.status = 'normal'
+    }
+    return seg;
+  });
+
+  markers = data.monitors.map(mon => {
+    return { id: mon.id, lat: mon.lat, lon: mon.long, status: mon.status }
+  });
+  return { markers, segments };
+}
+
+function generateMapIcons({ segments, markers },  simpleView) {
+  let defaultColor = simpleView ? backgroundColor : lighterColor;
+  return segments.map((segment) => {
+    let sensorIn = markers.find(marker => (marker.id === segment.senseIDIn));
```

Comment:
If you are not re-assigning then please use const. goes for the other variables as well

Reply:
I changed all my variables that are not re-assigned to const